modules/platforms/dotnet/Apache.Ignite/Table/ITableView.cs

```diff
@@ -31,15 +32,148 @@ public interface ITableView<T>
+
+        /// <summary>
+        /// Replaces a record with the same key columns if it exists.
+        /// </summary>
+        /// <param name="record">Record to insert.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a value indicating whether a record with the specified key was replaced.
+        /// </returns>
+        Task<bool> ReplaceAsync(T record);
+
+        /// <summary>
+        /// Replaces a record with a new one only if all existing columns have the same values
+        /// as the specified <paramref name="record"/>.
+        /// </summary>
+        /// <param name="record">Record to replace.</param>
+        /// <param name="newRecord">Record to replace with.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a value indicating whether a record was replaced.
+        /// </returns>
+        Task<bool> ReplaceAsync(T record, T newRecord);
+
+        /// <summary>
+        /// Replaces a record with the same key columns if it exists.
+        /// </summary>
+        /// <param name="record">Record to insert.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains the previous value for the given key, or <c>null</c> if it did not exist.
+        /// </returns>
+        Task<T?> GetAndReplaceAsync(T record);
+
+        /// <summary>
+        /// Deletes a record with the specified key.
+        /// </summary>
+        /// <param name="key">A record with key columns set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a value indicating whether a record with the specified key was deleted.
+        /// </returns>
+        Task<bool> DeleteAsync(T key);
+
+        /// <summary>
+        /// Deletes a record only if all existing columns have the same values as the specified <paramref name="record"/>.
+        /// </summary>
+        /// <param name="record">A record with all columns set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a value indicating whether a record was deleted.
+        /// </returns>
+        Task<bool> DeleteExactAsync(T record);
+
+        /// <summary>
+        /// Gets and deletes a record with the specified key.
+        /// </summary>
+        /// <param name="key">A record with key columns set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains deleted record or <c>null</c> if it did not exist.
+        /// </returns>
+        Task<T?> GetAndDeleteAsync(T key);
+
+        /// <summary>
+        /// Deletes multiple records.
```

Comment:
Again, we should probably clarify, whether the whole operation will fail if one of the records does not exist. I know it can be understand from the `<returns>` section, but I believe those details should be clear from the description itself.

Reply:
Fixed.